addons/onboarding/tests/test_onboarding.py

```diff
@@ -130,3 +130,31 @@ def test_onboarding_to_company_change(self):
         self.onboarding_1._search_or_create_progress()
 
         self.assert_onboarding_is_not_done(self.onboarding_1)
+
+    def test_no_crash_on_multiple_progress_records(self):
+        self.env.cr.execute(
```

Comment:
As you don't have a python constraint (yet), you could just create records using the python api directly?